<launch>

  <arg name="tf_name" default="" doc="nome del singolo robot per casi multirobot"/>

  <arg name="slash" value="$(eval '/' if tf_name!='' else '')" doc="argomento ausiliario per gestire tf frame"/>

  <node pkg="gmapping" type="slam_gmapping" name="turtlebot3_slam_gmapping">
    <!-- <remap from="scan" to="filtered_scan"/> -->
    <param name="odom_frame" value="$(arg slash)$(arg tf_name)$(arg slash)odom"/>
    <param name="base_frame" value="$(arg slash)$(arg tf_name)$(arg slash)base_footprint"/>
    <param name="map_frame"  value="$(arg slash)$(arg tf_name)$(arg slash)map"/>
    <rosparam command="load" file="$(find test_unknown_rendezvous)/params/gmapping_params.yaml" />
    <!-- --> 
    <remap from="scan" to="scan_filtered"/>
  </node>
  
  
  <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
    <rosparam command="load" file="$(find test_unknown_rendezvous)/params/lidar_config.yaml"/>
    <remap from="base_scan" to="scan"/>
  </node>


</launch>